data/tekton-pipelines/windows-customize-pipeline.yaml

```diff
@@ -0,0 +1,163 @@
+spec:
+  params:
+    - name: instanceType
+      type: string
+      description: "Name of VirtualMachineClusterInstancetype object"
+      default: "n1.large"
+    - name: preference
+      type: string
+      description: "Name of VirtualMachineClusterPreference object"
+      default: "windows.11"
+    - name: customizeConfigMapName
+      description: Name of the ConfigMap containing the customization and sysprep configuration files (unattend.xml, etc.). For example windows10-sqlserver or windows11-vs-code. It is possible to provide customize ConfigMaps created by the user too.
+      type: string
+      default: windows-vs-code
+    - name: sourceDiskImageName
+      description: Name of the windows source disk which will be copied and modified with sysprep
+      type: string
+      default: "win11"
+    - name: sourceDiskImageNamespace
+      description: Namespace of the windows source disk which will be copied and modified with sysprep
+      type: string
+      default: "kubevirt-os-images"
+  tasks:
+    - name: create-vm
+      params:
+        - name: runStrategy
+          value: RerunOnFailure
+        - name: "manifest"
+          value: |-          
+            apiVersion: kubevirt.io/v1
+            kind: VirtualMachine
+            metadata:
+              generateName: "windows-customize-automation-"
+            spec:
+              instancetype:
+                kind: VirtualMachineClusterInstancetype
+                name: n1.large
+              preference:
+                kind: VirtualMachineClusterPreference
+                name: windows.11
+              running: true
+              template:
+                spec:
+                  domain:
+                    devices:
+                      disks:
+                        - disk:
+                            bus: virtio
+                          name: rootdisk
+                          bootOrder: 1
+                        - cdrom:
+                            bus: sata
+                          name: sysprep
+                  volumes:
+                  - name: sysprep
+                    sysprep:
+                      configMap: 
+                        name: $(params.autounattendConfigMapName)
+                  - dataVolume:
+                      name:  $(tasks.copy-vm-root-disk.results.name)
+                    name: rootdisk
+    - name: wait-for-vmi-status
+      params:
+        - name: vmiName
+          value: $(tasks.create-vm.results.name)
+        - name: successCondition
+          value: status.phase == Succeeded
+        - name: failureCondition
+          value: status.phase in (Failed, Unknown)
+      runAfter:
+        - create-vm
+      timeout: 2h
+      taskRef:
+        kind: Task
+        name: wait-for-vmi-status
+    - name: create-base-dv
```

Comment:
Again, isn't the base-dv sufficient? I.e. cloning the source disk just once.

Reply:
Initially we have to copy a source disk as a root disk for VM (we should not modify source disk as it can be used for other pipelines). And the second copy is for copying root disk to destination namespace. We might not need the second copy and use directly the rootdisk as a result disk